File: src/main/java/com/hemanth/automation/pages/HomePage.java
package com.hemanth.automation.pages;

import com.hemanth.automation.config.ConfigReader;
import org.openqa.selenium.By;

public class HomePage extends BasePage {

    private final By productsLink = By.xpath("//a[contains(text(),'Products')]");
    private final By siteLogo = By.xpath("//img[@alt='Website for automation practice']");

    public void open() {
        driver.get(ConfigReader.getProperty("base.url"));
        dismissAdOverlayIfPresent();
    }

    public boolean isHomepageDisplayed() {
        return isDisplayed(siteLogo);
    }

    public void goToProductsPage() {
        dismissAdOverlayIfPresent();
        jsClick(productsLink);
    }
}

File: playwright-ts/pages/HomePage.ts
import { expect, Locator, Page } from '@playwright/test';

export class HomePage {
    readonly page: Page;
    readonly productsLink: Locator;
    readonly siteLogo: Locator;

    constructor(page: Page) {
        this.page = page;
        this.productsLink = page.locator("xpath=//a[contains(text(),'Products')]");
        this.siteLogo = page.locator("xpath=//img[@alt='Website for automation practice']");
    }

    async openHomePage() {
        await this.page.goto('/');
    }

    async verifyHomePageVisible() {
        await expect(this.siteLogo).toBeVisible();
    }

    async clickProducts() {
        await this.productsLink.click();
    }
}